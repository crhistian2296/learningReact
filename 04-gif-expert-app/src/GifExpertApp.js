import React, { useState } from "react";
import AddHero from "./components/AddHero";
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaultHero = ["Saitama"] }) => {
    // const heroes = ["saitama", "goku", "tanjiro"];
    const [heroes, setHeroes] = useState(defaultHero);

    // const handleAdd = () => {
    //     setHeroes((initialState) => [...initialState, "luffy"]);
    //     // console.log(typeof heroes);
    // };

    return (
        <>
            <h2 className="animate__animated animate__bounce">GifExpertApp</h2>
            <AddHero setHeroes={setHeroes} />
            <hr></hr>
            {/* <button type="submit" onClick={handleAdd}>
                add
            </button> */}

            {heroes.map((heroe) => (
                <GifGrid key={heroe} hero={heroe}></GifGrid>
            ))}
        </>
    );
};

export default GifExpertApp;
