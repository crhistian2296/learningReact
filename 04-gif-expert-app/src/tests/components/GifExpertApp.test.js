import { shallow } from "enzyme";
import React from "react";
import GifExpertApp from "../../GifExpertApp";

let wrapper = shallow(<GifExpertApp />);
beforeEach(() => {
    wrapper = shallow(<GifExpertApp />);
});

describe("Pruebas en el componente <GifExpertApp/>", () => {
    test("Debe renderizar el componente GifExpertApp correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Debe de mostrar una lista de heroes en busqueda", () => {
        const heroList = ["Papunika", "Hunterxhunter", "Deadpool"];
        wrapper = shallow(<GifExpertApp defaultHero={heroList} />);
        expect(wrapper.find("GifGrid").length).toBe(heroList.length);
        expect(wrapper).toMatchSnapshot();
    });
});
