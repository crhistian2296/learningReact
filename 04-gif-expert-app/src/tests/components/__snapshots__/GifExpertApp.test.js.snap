// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp/> Debe de mostrar una lista de heroes en busqueda 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__bounce"
  >
    GifExpertApp
  </h2>
  <AddHero
    setHeroes={[Function]}
  />
  <hr />
  <GifGrid
    hero="Papunika"
    key="Papunika"
  />
  <GifGrid
    hero="Hunterxhunter"
    key="Hunterxhunter"
  />
  <GifGrid
    hero="Deadpool"
    key="Deadpool"
  />
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp/> Debe renderizar el componente GifExpertApp correctamente 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__bounce"
  >
    GifExpertApp
  </h2>
  <AddHero
    setHeroes={[Function]}
  />
  <hr />
  <GifGrid
    hero="Saitama"
    key="Saitama"
  />
</Fragment>
`;
